completion/models.py

```diff
@@ -162,12 +162,26 @@ class BlockCompletion(TimeStampedModel, models.Model):
     id = BigAutoField(primary_key=True)  # pylint: disable=invalid-name
     user = models.ForeignKey(User)
     course_key = CourseKeyField(max_length=255)
+
+    # note: this usage key may not have the run filled in for
+    # old mongo courses.  Use the full_block_key property
+    # instead when you want to use/compare the usage_key.
     block_key = UsageKeyField(max_length=255)
     block_type = models.CharField(max_length=64)
     completion = models.FloatField(validators=[validate_percent])
 
     objects = BlockCompletionManager()
 
+    @property
+    def full_block_key(self):
```

Comment:
The logic here is being used in many different places. Would it make sense to extract this logic into a comment module like utils.py and use it everywhere this logic is needed, including here?

Reply:
Do you mean how this is being used in grades as well as completion?  Maybe it would make sense to make a Mixin class in opaque_keys?